inst/htmljs/animint.js

```diff
@@ -1766,6 +1771,17 @@ var animint = function (to_select, json_file) {
       // event will be fired on the original input.
       selectized_array[v_name].setValue(selected_ids, true);
     }
+
+    // For each updated geom, check if the axes of the plot need to be
+    // updated and update them
+    s_info.update.forEach(function(g_name){
+      var plot_name = g_name.split("_").pop();
+      var axes = Plots[plot_name]["options"]["update_axes"];
+      if(axes != null){
+        update_scales(plot_name, g_name, axes, v_name, value);
```

Comment:
@tdhock Sorry but could you please explain a bit? 
Currently I cycle through all the geoms in `s_info.update`. If the geom belongs to a plot that needs to have its axis updated, I take the appropriate ranges, and update the scales of the plot accordingly.  For eg, the geom_text will not have any effect on scales as it does not have a "state" selector, so we get `undefined` [here](https://github.com/tdhock/animint/pull/158/commits/a7370f4fb5a526d247a6ac7c962962ea20c3680a#diff-bfbd034b190fd160d92db9a24ea2a3c7R1733).
Then the `update_geom` function gets called as usual.

As far as I can understand, you are suggesting that I compute a single range value for a data subset for the whole plot in the compiler, which can be used to update the plot scales every time any geom belonging to it is updated. Is that correct?